nf_core/download.py

```diff
@@ -343,24 +547,32 @@ def prompt_compression_type(self):
         if self.compress_type == "none":
             self.compress_type = None
 
-    def download_wf_files(self):
+    def download_wf_files(self, revision, wf_sha, download_url):
         """Downloads workflow files from GitHub to the :attr:`self.outdir`."""
-        log.debug(f"Downloading {self.wf_download_url}")
+        log.debug(f"Downloading {download_url}")
 
         # Download GitHub zip file into memory and extract
-        url = requests.get(self.wf_download_url)
+        url = requests.get(download_url)
         with ZipFile(io.BytesIO(url.content)) as zipfile:
             zipfile.extractall(self.outdir)
 
+        # create a filesystem-safe version of the revision name for the directory
+        revision_dirname = re.sub("[^0-9a-zA-Z]+", "_", revision)
+        # account for name collisions, if there is a branch / release named "configs" or "singularity-images"
+        if revision_dirname in ["configs", "singularity-images"]:
+            revision_dirname = re.sub("[^0-9a-zA-Z]+", "_", self.pipeline + revision_dirname)
+
         # Rename the internal directory name to be more friendly
-        gh_name = f"{self.pipeline}-{self.wf_sha}".split("/")[-1]
-        os.rename(os.path.join(self.outdir, gh_name), os.path.join(self.outdir, "workflow"))
+        gh_name = f"{self.pipeline}-{wf_sha if bool(wf_sha) else ''}".split("/")[-1]
+        os.rename(os.path.join(self.outdir, gh_name), os.path.join(self.outdir, revision_dirname))
```

Comment:
No, `revision_dirname` is always a string, because the `revision` in `self.download_wf_files()` is too. 

But nice catch, it is indeed confusing with support for multiple revisions and no typing (_Promised, I will do that for downloads in a separate PR_).

```
for item in zip(self.revision, self.wf_sha.values(), self.wf_download_url.values()):
  revision_dirname = self.download_wf_files(revision=item[0], wf_sha=item[1], download_url=item[2])
```